webapp/components/create_post.jsx

```diff
@@ -179,8 +179,10 @@ class CreatePost extends React.Component {
         this.setState({messageText: '', submitting: false, postError: null, previews: [], serverError: null});
 
         Client.createPost(post,
-            () => {
-                // DO nothing. Websockets will handle this.
+            (data) => {
+                if (data) {
+                    PostStore.removePendingPost(data.pending_post_id);
```

Comment:
It would work with `post.pending_post_id` as well but I prefer to use the one provided in the argument to the callback that's coming back from the server. Unless you think that the server sometimes isn't returning the pending post id?